code/modules/mob/living/carbon/slime/slime.dm

```diff
@@ -204,6 +204,12 @@
 	updatehealth()
 	return
 
+/mob/living/carbon/slime/MouseDrop(var/atom/movable/A as mob|obj)
+	if(isliving(A) && A != usr)
+		var/mob/living/Food = A
+		if(Food.Adjacent(usr) && !stat && Food.stat != DEAD) //messy
```

Comment:
well unless i'm missing something i can't replace any of the three with canusetopic(), so it remains as is. if it's still a problem, i can close this and bring it back in a better form if i eventually get around to rewriting some slime procs.